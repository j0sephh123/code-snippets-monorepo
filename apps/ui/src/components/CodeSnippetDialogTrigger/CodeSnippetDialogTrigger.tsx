import { ActionIcon } from '@mantine/core';
import { IconPlus } from '@tabler/icons-react';
import { toggleDialog } from '../../store/dialog/dialogState';

export default function CodeSnippetDialogTrigger() {
  return (
    <ActionIcon
      onClick={() => toggleDialog('create', { title: 'Create a Code Snippet' })}
      style={{
        position: 'fixed',
        right: '20px',
        bottom: '20px',
      }}
      radius="xl"
      size="xl"
      variant="filled"
    >
      <IconPlus style={{ width: '70%', height: '70%' }} stroke={1.5} />
    </ActionIcon>
  );
}
