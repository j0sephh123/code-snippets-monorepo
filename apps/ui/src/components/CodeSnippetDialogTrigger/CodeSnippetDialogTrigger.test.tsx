import userEvent from '@testing-library/user-event';
import { screen } from '@testing-library/react';
import { vi } from 'vitest';
import CodeSnippetDialogTrigger from './CodeSnippetDialogTrigger';
import { renderWithProvider } from '../../tests/renderUtils';

const toggleDialogSpy = vi.hoisted(vi.fn);

vi.mock('../../store/dialog/dialogState', () => ({
  toggleDialog: toggleDialogSpy,
}));

describe('components > CodeSnippetDialogTrigger', () => {
  it('renders', async () => {
    const user = userEvent.setup();
    const { asFragment } = renderWithProvider(<CodeSnippetDialogTrigger />, {
      provider: 'mantine',
    });
    const btn = screen.getByRole('button');

    await user.click(btn);

    expect(toggleDialogSpy).toHaveBeenCalledWith('create', {
      title: 'Create a Code Snippet',
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
