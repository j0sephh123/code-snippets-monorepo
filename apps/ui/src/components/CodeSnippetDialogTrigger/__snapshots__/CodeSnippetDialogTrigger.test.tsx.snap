// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components > CodeSnippetDialogTrigger > renders 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <button
    aria-describedby=":r1:"
    class="mantine-focus-auto mantine-active _actionIcon_f46141 m-8d3f4000 mantine-ActionIcon-root m-87cf2631 mantine-UnstyledButton-root"
    data-size="xl"
    data-variant="filled"
    style="--ai-size: var(--ai-size-xl); --ai-radius: var(--mantine-radius-xl); --ai-bg: var(--mantine-color-blue-filled); --ai-hover: var(--mantine-color-blue-filled-hover); --ai-color: var(--mantine-color-white); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
    type="button"
  >
    <svg
      class="tabler-icon tabler-icon-plus"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
      style="width: 70%; height: 70%;"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 5l0 14"
      />
      <path
        d="M5 12l14 0"
      />
    </svg>
  </button>
</DocumentFragment>
`;
